### ISSUE TITLE
[HOLD for payment 2024-04-05] [$500] [Simplified Collect][Members] Changing owner: Full flow implementation

### ISSUE THREAD

**Issue by luacmartins** [Expensify/App#35716]
Implement `Changing owner: Full flow implementation` from the [Simplified Collect: Profile, Members, Categories design doc](https://docs.google.com/document/d/1gk3xqOs7epMbUrSSiX8K7YcqfPLVgqEos0sf-D-GMDA/edit)

<details><summary>Issue Owner</summary>Current Issue Owner: @burczu</details>
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01289ff00d54fea22e</li>
        <li>Upwork Job ID: 1755658801369747456</li>
        <li>Last Price Increase: 2024-02-08</li>
    </ul>
</details>

**Comment by luacmartins** [comment]
@burczu can you please comment here so I can assign you to the issue?

**Comment by burczu** [comment]
Hi! I'm Bartek from Callstack - expert contributor group. I'll work on this issue so please assign me. Thank you.

**Comment by CortneyOfstad** [comment]
@burczu any updates here? Thanks!

**Comment by burczu** [comment]
@CortneyOfstad this is one of the issues of the project and I'm working on these issues in an order - I didn't start working on this issue yet - I'll let you know here once started

**Comment by CortneyOfstad** [comment]
Sounds good @burczu!

**Comment by burczu** [comment]
not overdue - I didn't start the implementation yet

**Comment by trjExpensify** [comment]
Updated the title to reflect the hold to build the `WorkspaceMemberDetailsPage` first. ([Discussion ref](https://expensify.slack.com/archives/C036QM0SLJK/p1707768549059739))

**Comment by CortneyOfstad** [comment]
Thanks @trjExpensify!

**Comment by CortneyOfstad** [comment]
Linking the on-hold issue here for easier access — https://github.com/Expensify/App/issues/35715

#35715 is waiting for for PR implementation, and I'll keep an eye to see when that gets rolling to move this one along as well 👍

**Comment by CortneyOfstad** [comment]
The on-hold issue is still open, so no update here. Going to continue to keep this as `daily` as that is what https://github.com/Expensify/App/issues/35715 is set to as well.

**Comment by CortneyOfstad** [comment]
@burczu bump [here](https://github.com/Expensify/App/issues/35716#issuecomment-1964253978). Can you please provide an update by EOD?

**Comment by CortneyOfstad** [comment]
Hi @adelekennedy! I am heading OoO and will be back March 11th, so reassigned to have someone keep this moving until I am back. At this stage, @burczu is working on this, so just wanted to have eyes on it just in case something was needed while I was gone. Thanks!

**Comment by burczu** [comment]
@CortneyOfstad @adelekennedy @luacmartins As [the blocking issue](https://github.com/Expensify/App/issues/35715) has the PR ready for review, I'm starting the implementation of this issue (while addressing potential feedback for the PR, of course) - I think we can remove ON HOLD here.

I'll create the new branch on top of the branch I've used for https://github.com/Expensify/App/issues/35715 implementation, so I think the most reasonable will be that I'll post the draft PR once the other PR is merged - then, only changes related to this issue will be visible there. Is it ok?

**Comment by luacmartins** [comment]
Yes, let's start work on this one while we review https://github.com/Expensify/App/issues/35715. I think we can post the draft PR while still reviewing the other one, we won't really review anything for the new PR, it's just a way to track progress.

**Comment by burczu** [comment]
Draft created.

**Comment by burczu** [comment]
Daily updated: today I was mostly busy addressing some regressions from my previous PR, but till now I was able to make some changes to allow pressing on the row in the Members list while preventing from pressing the checkbox at the same time (this is necessary for some constraints in bulk actions - discussed this with @luacmartins today). I've also added the the `Transfer owner` option visible for admins that are not owners, when they open owner's details. From this state I'll be able to add next steps of the change owner flow.

**Comment by luacmartins** [comment]
Nice! Thanks for the updated!

**Comment by CortneyOfstad** [comment]
Back from OoO — thanks for keeping an eye on things @adelekennedy!

**Comment by burczu** [comment]
Daily update: I've created pages for all the change ownership confirmations and partially for the add payment card form - I think the core functionality for this feature should be done tomorrow.

**Comment by CortneyOfstad** [comment]
Sounds great! Thanks @burczu!

**Comment by luacmartins** [comment]
Sorry for the delay here. @burczu reached out asking for some screen copies, but I didn't get this back to him in time yesterday. I've replied today and I think he has all the copy needed here.

**Comment by luacmartins** [comment]
I think @burczu is out for the day already. We're discussing this 1:1 since there are a lot of different pages/steps to cover.

**Comment by CortneyOfstad** [comment]
Sounds good @luacmartins! Looking forward to seeing what you and @burczu come up with!

**Comment by luacmartins** [comment]
@CherylWalsh we currently have the modal below in OldDot. This modal is displayed when a user is trying to transfer the ownership of a workspace to themselves, but there's an unknown error clearing the owed balance. Could you please check if we want to keep the same working in NewDot?

![Screenshot 2024-03-20 at 10 08 30 AM](https://github.com/Expensify/App/assets/22219519/7144484c-2173-473c-b9db-9c5b3bc38cbe)

**Comment by CortneyOfstad** [comment]
@mollfpr — can you update the checklist before EOD tomorrow if a regression test is needed? Thanks!

**Comment by luacmartins** [comment]
No need to add tests here. We're adding it as part of the project - https://github.com/Expensify/Expensify/issues/383378

**Comment by CortneyOfstad** [comment]
## Payment Summary

@mollfpr to be paid $500 via NewDot

**Comment by JmillsExpensify** [comment]
$500 approved for @mollfpr
